packages/core/core/src/RequestTracker.js

```diff
@@ -1108,75 +1115,124 @@ export default class RequestTracker {
     return {api, subRequestContentKeys};
   }
 
-  async writeToCache() {
+  async writeToCache(signal?: AbortSignal) {
     let cacheKey = getCacheKey(this.options);
-    let requestGraphKey =
-      hashString(`${cacheKey}:requestGraph`) + '-RequestGraph';
-    let snapshotKey = hashString(`${cacheKey}:snapshot`);
+    let requestGraphKey = `requestGraph-${hashString(cacheKey)}`;
+    let snapshotKey = `snapshot-${hashString(cacheKey)}`;
 
     if (this.options.shouldDisableCache) {
       return;
     }
-    let total = 2;
+
+    let serialisedGraph = this.graph.serialize();
+
+    let total = 0;
+    const serialiseAndSet = async (
+      key: string,
+      // $FlowFixMe serialise input is any type
+      contents: any,
+    ): Promise<void> => {
+      if (signal?.aborted) {
+        throw new Error('Serialization was aborted');
+      }
+
+      await this.options.cache.setLargeBlob(
+        key,
+        serialize(contents),
+        signal
+          ? {
+              signal: signal,
+            }
+          : undefined,
+      );
+
+      total += 1;
+
+      report({
+        type: 'cache',
+        phase: 'write',
+        total,
+        size: this.graph.nodes.length,
+      });
+    };
+
+    let queue = new PromiseQueue({
+      maxConcurrent: 32,
+    });
+
     report({
       type: 'cache',
       phase: 'start',
       total,
       size: this.graph.nodes.length,
     });
-    let promises = [];
-    for (let node of this.graph.nodes) {
-      if (!node || node.type !== REQUEST) {
+
+    // Preallocating a sparse array is faster than pushing when N is high enough
+    let cacheableNodes = new Array(serialisedGraph.nodes.length);
+    for (let i = 0; i < serialisedGraph.nodes.length; i += 1) {
+      let node = serialisedGraph.nodes[i];
+
+      let resultCacheKey = node?.resultCacheKey;
+      if (
+        node?.type === REQUEST &&
+        resultCacheKey != null &&
+        node?.result != null
+      ) {
+        queue
+          .add(() => serialiseAndSet(resultCacheKey, node.result))
+          .catch(() => {});
+        // eslint-disable-next-line no-unused-vars
+        let {result: _, ...newNode} = node;
+        cacheableNodes[i] = newNode;
+      } else {
+        cacheableNodes[i] = node;
+      }
+    }
+
+    for (let i = 0; i * NODES_PER_BLOB < cacheableNodes.length; i += 1) {
+      if (
+        this.cachedRequestsLastChunk !== null &&
+        i < this.cachedRequestsLastChunk
```

Comment:
The assumption here is that nodes never change or get re-ordered?

Reply:
Yep added a comment here. This idea was from @devongovett in the sync